src/enclave/rpcendpoint.h

```diff
@@ -34,31 +35,113 @@ namespace enclave
       session_id(session_id)
     {}
 
+    std::optional<jsonrpc::Pack> detect_pack(const std::vector<uint8_t>& input)
```

Comment:
We should make sure that `detect_pack()` and `unpack_json()` are not duplicated from the ones in `frontend.h` before we merge this.

Reply:
Yes, absolutely, I'm removing a load of code from the frontend at the moment.